Bluetooth: Controller: Remove experimental for adv and periodic adv

cvinayak opened the pull request:
Remove experimental from Advertising Extensions and Periodic
Advertising features.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Remove experimental from Ext and Periodic Adv

Remove experimental label from Advertising Extensions and
Periodic Advertising features in the Controller.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley reviewed (commented):
This is cause for celebration! 

Should the samples and tests that explicitly set `BT_CTLR_ADV_EXT` etc. be updated to no longer set `BT_CTLR_ADV_EXT` if `BT_EXT_ADV` is set?

cvinayak committed:
samples: Bluetooth: Remove explicit Ext and Periodic Adv Kconfig enable

Remove the explicit enable of Extended and Periodic
Advertising support in Controller, the feature is no
longer experimental in the Controller and is now enabled
by default when application uses them by enabling as
Host feature.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
> This is cause for celebration!
> 
> Should the samples and tests that explicitly set `BT_CTLR_ADV_EXT` etc. be updated to no longer set `BT_CTLR_ADV_EXT` if `BT_EXT_ADV` is set?

Added a new commit to remove board specific configuration files that earlier had explicit enable of Extended and Periodic Advertising for the Controller.

Tests that do not use board specific configuration, direction finding and audio related sample which are still experimental are not changed in this PR (those be clean up when they become non-experimental).
